akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/CommandHandler.scala

```diff
@@ -62,7 +61,7 @@ final class CommandHandlerBuilder[Command, Event, S <: State, State] @InternalAp
   private def addCase(predicate: Command ⇒ Boolean, handler: BiFunction[S, Command, Effect[Event, State]]): Unit = {
     cases = CommandHandlerCase[Command, Event, State](
       commandPredicate = predicate,
-      statePredicate = state ⇒ stateClass.isAssignableFrom(state.getClass) && statePredicate.test(state.asInstanceOf[S]),
+      statePredicate = state ⇒ statePredicate.test(state.asInstanceOf[S]) && stateClass.isAssignableFrom(state.getClass),
```

Comment:
we need to first pass through the predicate, otherwise we can support null states